nixos/modules/services/misc/transfer-sh.nix

```
@@ -0,0 +1,434 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.transfer-sh;
+in
+{
+  options = {
+    services.transfer-sh = {
+
+      enable = mkEnableOption "transfer-sh setup";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.tansfer-sh;
+        defaultText = literalExpression "pkgs.transfer-sh";
+        description = lib.mdDoc "transfer-sh derivation to use.";
+      };
+
+      environmentFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = lib.mdDoc "Environment file as defined in {manpage}`systemd.exec(5)`.";
+      };
+
+      user = mkOption {
+        description = "user to run as";
+        default = "transfersh";
+        type = types.str;
+      };
+
+      group = mkOption {
+        description = "group to run as";
+        default = "transfersh";
+        type = types.str;
+      };
+
+      provider = mkOption {
+        description = "which storage provider to use (s3, storj, gdrive or local)";
+        default = "local";
+        type = types.enum [ "s3" "storj" "gdrive" "local" ];
+      };
+
+      address = mkOption {
+        description = "address to listen on";
+        default = "127.0.0.1";
+        type = types.string;
+      };
+
+      LISTENER = mkOption {
+        description = "port to use for http";
+        default = 6080;
+        type = types.nullOr types.int;
+      };
+
+      PROFILE_LISTENER = mkOption {
+        description = "port to use for profiler";
+        default = 6060;
+        type = types.nullOr types.int;
```

```suggestion
        type = types.port;
      };

      PROFILE_LISTENER = mkOption {
        description = "port to use for profiler";
        default = 6060;
        type = types.nullOr types.port;
```
I assume we always need at least the http port